packages/strapi-admin/services/permission.js

```diff
@@ -180,6 +189,43 @@ const cleanPermissionInDatabase = async () => {
   }
 };
 
+const ensureBoundPermissionsInDatabase = async () => {
+  if (strapi.EE) return;
+
+  const models = Object.values(strapi.contentTypes);
+  const role = await strapi.query('role', 'admin').findOne({ code: EDITOR_CODE }, []);
+
+  for (const model of models) {
+    const actions = getBoundActionsBySubject(role, model.uid);
+    const perms = await strapi.query('permission', 'admin').find(
+      {
+        subject: model.uid,
+        action_in: actions,
+        role: role.id,
+      },
+      []
+    );
+
+    if (perms.length > 0) {
+      const missingActions = _.difference(actions, _.map(perms, 'action'));
+
+      if (missingActions.length > 0) {
+        const { fields } = perms[0];
+
+        const permissions = missingActions.map(action =>
+          createPermission({
+            action,
+            subject: model.uid,
+            role: role.id,
+            fields: BOUND_ACTIONS_FOR_FIELDS.includes(action) ? fields : null,
+          })
+        );
+        await createMany(permissions);
+      }
+    }
+  }
+};
```

Comment:
It all works if a permission action is missing but not if a field is missing for a permission action. It would not required much more work to handle this case. What do you think?

Reply:
that's already handled in your clean permission

Reply:
I'm not sure. I checked and I don't see where this case is handled

Reply:
It's handled in cleanPermissionInDatabase, I just checked

Reply:
![image](https://user-images.githubusercontent.com/25851739/95077167-51053180-0713-11eb-9e81-f80dc891832e.png)

Reply:
Currently only required fields and super admin's permissions fields are added if needed.
We still need to add the missing & non-required ones for author & editor roles in CE.